First delivery

vesin opened the pull request.

vesin committed:
Initial commit

Coordinates conversion tool have bug

vesin committed:
Doc fix.

vesin committed:
TODO fixes and small refactoring.

vesin committed:
TODO cleanup and small refactoring.

cikaRadule committed:
Fixed location converter

cikaRadule committed:
Fixed coordinates converter tool.

woutslakhorst commented on README.md:
```diff
@@ -1,4 +1,43 @@
+Necessary properties files are the same as in DEVELOPMENT instruction part with minor difference that some of the files dont have "_tests" part in its name. Name of the files shoud be db.properties, geocoding.properties and log4j.properties.
+
+#Development instruction
+To start with development every developer need to create configuration a folder for test properties in api/config/test/{username} 
```
This is strange to have in a open-source project. Please remove. Have the necessary files for test in something like src/test/resources (mvn default) and user GEO_CODING_HOME otherwise. So remove all the developer specific dirs.

cikaRadule replied:
Can you please confirm one thing:
You want both cong/test/{user.name} and GEO_CODING_HOME properties out of the project?
(we should remove developer specific dirs from main and from test)

woutslakhorst replied:
GEO_CODING_HOME should be the only mechanism left for development and production. Test should use the config files in test. (src/test/resources). For running a server locally we can use -DGEO_CODING_HOME=something. So don't remove GEO_CODING_HOME

woutslakhorst commented:
Culd you add coverage plugins like cobertura to the pom?

cikaRadule committed:
Modified api pom, war is deployed and tested.

woutslakhorst commented on api/src/main/java/com/nedap/healthcare/kadasterbagclient/api/model/AbstractPersistedEntity.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    @Override
+    public int hashCode() {
```
Why not just id? It's the primary key so uniqueness is garanteed.

woutslakhorst commented on api/src/main/java/com/nedap/healthcare/kadasterbagclient/api/model/Address.java:
```diff
@@ -0,0 +1,411 @@
+
+    /** House number. */
+    @Column(nullable = false)
+    private Integer number;
```
This reminds me (the integer triggered me), we need a postfix column to support numbers like 2a.

woutslakhorst commented on api/src/main/java/com/nedap/healthcare/kadasterbagclient/api/model/basic/CalendarType.java:
```diff
@@ -0,0 +1,144 @@
+ * 
+ * @author dvesin
+ */
+public class CalendarType extends MutableType implements VersionType {
```
I'm guessing this is used for mapping a calendar on the DB? Scrap it and use the Joda DateTime class.

woutslakhorst commented on api/src/main/java/com/nedap/healthcare/kadasterbagclient/api/service/LocationServiceImpl.java:
```diff
@@ -0,0 +1,176 @@
+
+        if (location != null && isExpired(location)) {
+            locationDao.delete(location);
+            location = locationDao.findByCountryPostalCodeAndNumber(NL_COUNTRY_CODE, zipCode, houseNumber);
```
why not: 
location = null;

cikaRadule replied:
Hibernate fails to remove entity from db if second call of "findByCountryPostalCodeAndNumber" is omitted.
ConstraintViolationException is raised on next insert for expired location data.

woutslakhorst replied:
That's strange. Could that have something to do with the fact the session needs to be purged?

woutslakhorst commented on eclipse/execomCheckstyle.xml:
```diff
@@ -0,0 +1,107 @@
```
Could you add the eclipse dir to the gitignore file?

woutslakhorst commented on model/src/main/java/com/nedap/healthcare/kadasterbagclient/model/AddressDTO.java:
```diff
@@ -0,0 +1,116 @@
```
This class will also need the POSTFIX String variable

woutslakhorst commented on api/pom.xml:
```diff
@@ -0,0 +1,235 @@
+		</dependency>
+
+		<dependency>
+			<groupId>eu.execom</groupId>
```
Is this testutil available in public repo's? It's currently packaged with the source which is kind of ugly.

woutslakhorst commented:
mvn test doesn't work out of the box
doing a mvn test on the service gives:

mvn test
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.nedap.healthcare.kadasterbagclient:service:jar:0.0.1-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 59, column 12
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO]  
[INFO] ------------------------------------------------------------------------
[INFO] Building service 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- cxf-codegen-plugin:2.2.9:wsdl2java (test-service) @ service ---
[INFO] 
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ service ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Volumes/nedap_healthcare/kadaster-bag-client/service/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.3.2:compile (default-compile) @ service ---
[INFO] Compiling 100 source files to /Volumes/nedap_healthcare/kadaster-bag-client/service/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Volumes/nedap_healthcare/kadaster-bag-client/service/target/generated/nl/kadaster/schemas/imbag/apd/v20090901/Verblijfsobject.java:[23,49] unmappable character for encoding UTF-8
[ERROR] /Volumes/nedap_healthcare/kadaster-bag-client/service/target/generated/nl/kadaster/schemas/imbag/apd/v20090901/Verblijfsobject.java:[23,50] unmappable character for encoding UTF-8
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.860s
[INFO] Finished at: Tue Sep 11 09:20:16 CEST 2012
[INFO] Final Memory: 6M/265M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project service: Compilation failure: Compilation failure:
[ERROR] /Volumes/nedap_healthcare/kadaster-bag-client/service/target/generated/nl/kadaster/schemas/imbag/apd/v20090901/Verblijfsobject.java:[23,49] unmappable character for encoding UTF-8
[ERROR] /Volumes/nedap_healthcare/kadaster-bag-client/service/target/generated/nl/kadaster/schemas/imbag/apd/v20090901/Verblijfsobject.java:[23,50] unmappable character for encoding UTF-8
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

woutslakhorst commented:
remove the doc folder. This is information for Nedap and Execom, not for the public.

woutslakhorst commented:
Merge the API project with the Service project. We're probably not going to use the separate API.

woutslakhorst commented:
service sources seem to be generated in target/generated. This should be in target/generated-sources/

cikaRadule committed:
Changed remote connection to web service.
Tested on local machine. Couple of tests should be added.

cikaRadule committed:
Testing web service changed.
Access to web service while running test cases is changed to work throu http request.
Added new test for the case when web service is not available and client tries to access it.

cikaRadule commented:
I don't see reply on comment option here, so let's try like this...
"Merge the API project with the Service project. We're probably not going to use the separate API."
Can you just clarify, what module should be merget into the other?
In other words, after merge what module should prevail?

cikaRadule committed:
Service code generating target changed.
Generated code moved from "generated" to "generated-sources" folder.

cikaRadule committed:
Removed doc and eclipse user specific files.

woutslakhorst commented:
Service should prevail.

cikaRadule committed:
Removed hash code function from Address and AbstractPersistedEntity

vesin commented:
On our last meeting you mention that Model should be merged into API also, so now we will have only one project?

woutslakhorst commented:
Sorry, I was a bit confused with the names. Model should be merged into API. Two projects remain Service and API. It's logical to have the DTO in the API

cikaRadule commented:
I've already merged api into service...
Bad luck :)

woutslakhorst commented:
Sorry, but with the right source control, rolling back is easy :)

cikaRadule committed:
Number postfix added to Address object.

cikaRadule committed:
Model merged into api module.

cikaRadule committed:
Changed location for test configuration files.

vesin committed:
Switching to latest hibernate version 4.1.6 and adding joda time
instead of Calendar.

cikaRadule committed:
Added Cobertura plugin.

cikaRadule committed:
Merge branch 'master' of https://github.com/vesin/kadaster-bag-client

Conflicts:
	api/pom.xml
	api/src/main/java/com/nedap/healthcare/kadasterbagclient/api/model/Address.java
	api/src/test/java/com/nedap/healthcare/kadasterbagclient/api/dao/LocationDaoTest.java
	api/src/test/java/com/nedap/healthcare/kadasterbagclient/api/service/LocationServiceTest.java

vesin committed:
remove old model dependency from pom.xml

vesin committed:
Revert hibernate to validate

Commit:
Removed model from api module.

Commit:
Fixed delete action in AbstractPersistEntity class.

Commit:
Merge branch 'master' of https://github.com/vesin/kadaster-bag-client
